dspace-api/src/main/java/org/dspace/xmlworkflow/state/actions/processingaction/SelectReviewerAction.java

```diff
@@ -67,49 +79,93 @@ public ActionResult execute(Context c, XmlWorkflowItem wfi, Step step, HttpServl
-            return new ActionResult(ActionResult.TYPE.TYPE_PAGE, SEARCH_RESULTS_PAGE);
-        } else if (submitButton.startsWith(SUBMIT_SELECT_REVIEWER)) {
-            //Retrieve the identifier of the eperson which will do the reviewing
-            UUID reviewerId = UUID.fromString(submitButton.substring(submitButton.lastIndexOf("_") + 1));
-            EPerson reviewer = ePersonService.find(c, reviewerId);
-            //Assign the reviewer. The workflowitemrole will be translated into a task in the autoassign
+    /**
+     * Method to handle the "submit_select_reviewer" action
+     *
+     * @param c       current DSpace session
+     * @param wfi     the item on which the action is to be performed
+     * @param step    the workflow step in which the action is performed
+     * @param request the current client request
+     * @return the result of performing the action
+     * @throws SQLException
+     * @throws AuthorizeException
+     */
+    public ActionResult processSelectReviewers(Context c, XmlWorkflowItem wfi, Step step, HttpServletRequest request)
+        throws SQLException, AuthorizeException {
+        //Retrieve the identifier of the eperson which will do the reviewing
+        String[] reviewerIds = request.getParameterValues("eperson");
+        if (ArrayUtils.isEmpty(reviewerIds)) {
+            return new ActionResult(ActionResult.TYPE.TYPE_ERROR);
+        }
+        List<EPerson> reviewers = new ArrayList<>();
+        for (String reviewerId : reviewerIds) {
+            reviewers.add(ePersonService.find(c, UUID.fromString(reviewerId)));
+        }
+        if (reviewers.size() == 1) {
+            if (!groupService.allMembers(c, getGroup()).contains(reviewers.get(0))) {
+                return new ActionResult(ActionResult.TYPE.TYPE_ERROR);
+            }
+            //We have a reviewer, assign him, the workflowitemrole will be translated into a task in the autoassign
+            WorkflowItemRole workflowItemRole = workflowItemRoleService.create(c);
+            workflowItemRole.setEPerson(reviewers.get(0));
+            workflowItemRole.setRoleId(getRole().getId());
+            workflowItemRole.setWorkflowItem(wfi);
+            workflowItemRoleService.update(c, workflowItemRole);
+            return new ActionResult(ActionResult.TYPE.TYPE_OUTCOME, ActionResult.OUTCOME_COMPLETE);
+        } else {
+            if (!groupService.allMembers(c, getGroup()).containsAll(reviewers)) {
+                return new ActionResult(ActionResult.TYPE.TYPE_ERROR);
+            }
+            //We have multiple reviewers, create a group and assign this group, the workflowitemrole will be
+            // translated into a task in the autoassign
+            c.turnOffAuthorisationSystem();
+            Group reviewerGroup = groupService.create(c);
+            for (EPerson reviewer : reviewers) {
+                groupService.addMember(c, reviewerGroup, reviewer);
+            }
+            c.restoreAuthSystemState();
```

Comment:
No this was the group created from the multiple selected reviewers for the `WorkflowItemRole` which makes claimed tasks for reviewers in `AutoAssignAction`, but indeed exposes the issue that these groups didn't get cleaned up after. Added a boolean config `deleteTemporaryGroup` to `Role` so that these get cleaned up after the auto-assign if the config is set to true.